modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/BloggingPageModel.cs

```
@@ -8,5 +12,26 @@ public BloggingPageModel()
         {
             ObjectMapperContext = typeof(BloggingWebModule);
         }
+        
+        protected async Task<BlogDto> GetBlogAsync(IBlogAppService blogAppService, BloggingUrlOptions blogOptions, string blogShortName)
+        {
+            if (!blogOptions.SingleBlogMode.Enable)
+            {
+                return await blogAppService.GetByShortNameAsync(blogShortName);
+            }
+
+            if (!blogOptions.SingleBlogMode.BlogName.IsNullOrEmpty())
+            {
+                return await blogAppService.GetByShortNameAsync(blogOptions.SingleBlogMode.BlogName);
+            }
+
+            var blogs = await blogAppService.GetListAsync();
+            if (blogs.Items.Count == 1)
+            {
+                return blogs.Items[0];
+            }
+            
+            return await blogAppService.GetByShortNameAsync(blogOptions.SingleBlogMode.BlogName);
```

Here, I think we don't need to get all blog posts. Instead, you can get a single blog by creationTime, or throw entity not found exception if there is not any blog in the system. Because, in line 34, we know the **blogOptions.SingleBlogMode.BlogName** is null, so no need to make a call for that.